tests/kafkatest/services/mirror_maker.py

```diff
@@ -0,0 +1,173 @@
+        """
+        MirrorMaker mirrors messages from one or more source clusters to a single destination cluster.
+
+        Args:
+            context:                    standard context
+            sources:                    list of one or more source kafka clusters
+            target:                     target cluster
+            whitelist:                  whitelist regex for topics to mirror
+            blacklist:                  blacklist regex for topics not to mirror
+            num_streams:                number of consumer threads to create
+            consumer_timeout_ms:        consumer stops if t > consumer_timeout_ms elapses between consecutive messages
+        """
+        super(MirrorMaker, self).__init__(context, num_nodes=1)
+
+        self.consumer_timeout_ms = consumer_timeout_ms
+        self.num_streams = num_streams
+        self.whitelist = whitelist
+        self.blacklist = blacklist
+        self.sources = sources
+        self.target = target
+
+    def consumer_config_file(self, kafka):
+        return "consumer%s.properties" % kafka.zk.connect_setting()
+
+    @property
+    def start_cmd(self):
+        cmd = "export LOG_DIR=%s;" % MirrorMaker.LOG_DIR
+        cmd += " export KAFKA_LOG4J_OPTS=\"-Dlog4j.configuration=file:%s\";" % MirrorMaker.LOG4J_CONFIG
+        cmd += " %s/bin/kafka-run-class.sh kafka.tools.MirrorMaker" % MirrorMaker.KAFKA_HOME
+        for kafka in self.sources:
+            # One consumer config file per source kafka cluster
+            cmd += " --consumer.config %s" % self.consumer_config_file(kafka)
+        cmd += " --producer.config %s" % MirrorMaker.PRODUCER_CONFIG
+        cmd += " --num.streams %d" % self.num_streams
+
+        if self.whitelist is not None:
+            cmd += " --whitelist=%s" % self.whitelist
+        if self.blacklist is not None:
+            cmd += " --blacklist=%s" % self.blacklist
+        cmd += " 2>&1 1>>%s &" % MirrorMaker.LOG_FILE
+        return cmd
+
+    def pids(self, node):
+        try:
+            cmd = "ps ax | grep -i MirrorMaker | grep java | grep -v grep | awk '{print $1}'"
+            pid_arr = [pid for pid in node.account.ssh_capture(cmd, allow_fail=True, callback=int)]
+            return pid_arr
+        except:
+            return []
+
+    def alive(self, node):
+        return len(self.pids(node)) > 0
+
+    def _worker(self, idx, node):
+        node.account.ssh("mkdir -p %s" % MirrorMaker.PERSISTENT_ROOT, allow_fail=False)
+        node.account.ssh("mkdir -p %s" % MirrorMaker.LOG_DIR, allow_fail=False)
+
+        # Create, upload one consumer config file per source kafka cluster
+        for kafka in self.sources:
+            consumer_props = self.render('consumer.properties', zookeeper_connect=kafka.zk.connect_setting(),
+                                         consumer_timeout_ms=self.consumer_timeout_ms)
```

Comment:
You shouldn't need to pass in `consumer_timeout_ms` like this -- since it's a field on the object calling `render`, it should already be available to the template.